server/src/main/java/io/seata/server/cluster/raft/RaftServerFactory.java

```diff
@@ -0,0 +1,232 @@
+            if (autoJoinCluster) {
+                List<PeerId> currentPeers = null;
+                try {
+                    currentPeers = getCliServiceInstance().getPeers(group, initConf);
+                } catch (Exception e) {
+                    // In the first deployment, the leader cannot be found
+                }
+                if (CollectionUtils.isNotEmpty(currentPeers) && !currentPeers.contains(serverId)) {
+                    Status status = getCliServiceInstance().addPeer(group, initConf, serverId);
+                    if (!status.isOk()) {
+                        LOGGER.error(
+                            "failed to join the RAFT cluster: {}. Please check the status of the cluster, status error: {}",
+                            initConfStr, status.getErrorMsg());
+                    }
+                }
+            }
+            LOGGER.info("started seata server raft cluster, group: {} ", group);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("fail init raft cluster:" + e.getMessage());
+        }
+    }
+
+    public RaftServer getRaftServer(String group) {
+        return RAFT_SERVER_MAP.get(group);
+    }
+
+    public Collection<RaftServer> getRaftServers() {
+        return RAFT_SERVER_MAP.values();
+    }
+
+    public Boolean isLeader(String group) {
+        AtomicReference<RaftStateMachine> stateMachine = new AtomicReference<>();
+        Optional.ofNullable(RAFT_SERVER_MAP.get(group)).ifPresent(raftServer -> {
+            stateMachine.set(raftServer.getRaftStateMachine());
+        });
+        RaftStateMachine raftStateMachine = stateMachine.get();
+        return !isRaftMode() && RAFT_SERVER_MAP.isEmpty() || (raftStateMachine != null && raftStateMachine.isLeader());
+    }
+
+    public Boolean isRaftMode() {
+        return raftMode;
+    }
+
+    private RaftOptions initRaftOptions() {
+        RaftOptions raftOptions = new RaftOptions();
+        raftOptions.setApplyBatch(CONFIG.getInt(SERVER_RAFT_APPLY_BATCH, raftOptions.getApplyBatch()));
+        raftOptions.setMaxAppendBufferSize(
+            CONFIG.getInt(SERVER_RAFT_MAX_APPEND_BUFFER_SIZE, raftOptions.getMaxAppendBufferSize()));
+        raftOptions.setDisruptorBufferSize(
+            CONFIG.getInt(SERVER_RAFT_DISRUPTOR_BUFFER_SIZE, raftOptions.getDisruptorBufferSize()));
+        raftOptions.setMaxReplicatorInflightMsgs(
+            CONFIG.getInt(SERVER_RAFT_MAX_REPLICATOR_INFLIGHT_MSGS, raftOptions.getMaxReplicatorInflightMsgs()));
+        raftOptions.setSync(CONFIG.getBoolean(SERVER_RAFT_SYNC, raftOptions.isSync()));
+        return raftOptions;
+    }
+
+    private NodeOptions initNodeOptions(Configuration initConf) {
+        NodeOptions nodeOptions = new NodeOptions();
+        // enable the CLI service.
+        nodeOptions.setDisableCli(false);
+        // snapshot should be made every 30 seconds
+        int snapshotInterval = CONFIG.getInt(SERVER_RAFT_SNAPSHOT_INTERVAL, 60 * 10);
```

Comment:
Change the Default value or change the comment

Reply:
done